vars/buildUnityWdkV4.groovy

```diff
@@ -0,0 +1,296 @@
+#!/usr/bin/env groovy
+import net.wooga.jenkins.pipeline.check.steps.Step
+import net.wooga.jenkins.pipeline.config.PipelineConventions
+import net.wooga.jenkins.pipeline.config.Platform
+import net.wooga.jenkins.pipeline.config.WDKConfig
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                                                                                                                    //
+// Step buildUnityWdkV4                                                                                               //
+//                                                                                                                    //
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+def call(Map configMap = [unityVersions: []]) {
+  def defaultReleaseType = "snapshot"
+  def defaultReleaseScope = ""
+
+  configMap.logLevel = configMap.get("logLevel", params.LOG_LEVEL ?: env.LOG_LEVEL as String)
+  configMap.showStackTrace = configMap.get("showStackTrace", params.STACK_TRACE as Boolean)
+  configMap.refreshDependencies = configMap.get("refreshDependencies", params.REFRESH_DEPENDENCIES as Boolean)
+  configMap.clearWs = configMap.get("clearWs", params.CLEAR_WS as boolean)
+  configMap.testWrapper = { Step testOperation, Platform plat ->
+    withCredentials([file(credentialsId: 'atlas-upm-credentials', variable: "UPM_USER_CONFIG_FILE")]) {
+      testOperation(plat)
+    }
+  }
+  def config = WDKConfig.fromConfigMap(configMap, this)
+  def hasToPublishAutoref = config.unityVersions.any({ version -> version.autoref })
```

Comment:
`autoref` shouldn't be a property of `unityVersions` if its independent from it, and its triggered when any version is set. It should be a property of the `WDKConfig` class.